web/src/utils.js

```diff
@@ -7,3 +7,22 @@ OS.metaKey = function metaKey() {
 };
 
 export default OS;
+
+const dustBase = '/dust/';
+const apiBase = '/api/v1';
+const baseUrl = '/maiden';
+
+export const findSubPath = (dustPath) => {
+  const i = dustPath.indexOf(dustBase);
+  return ~i ? dustPath.substring(i + dustBase.length) : null;
+}
+
+export const resourceToPath = (resource) => {
+  const sub = findSubPath(resource);
+  return sub ? `${baseUrl}${dustBase}${sub}` : null;
+}
+
+export const pathToResource = (path) => {
+  const sub = findSubPath(path);
+  return sub ? `${apiBase}${dustBase}${sub}` : null;
+}
```

Comment:
These constants partly duplicate values which are already defined in the `api.js` file. It would be preferable to use those:
https://github.com/monome/maiden/blob/aff75f1243451948b0e2053fb13c22dfb77d2618/web/src/api.js#L3-L13

It would be worth using (and refactoring as needed) the various static functions in api.js which are doing similar operations; for example functions like `resourceForScript(...)` and `fileFromResource(...)` seem similar-ish.

Reply:
Oh, nice, that's really helpful. This is my first time poking around in here so I'm not surprised I glossed over existing constants.

Reply:
I exported the `API_ROOT` constant here. I'm not totally seeing the forest for the trees with the refactoring of those functions yet. I might need some more concrete guidance there if more changes are needed.

Reply:
OK I think I see what you meant now. I didn't refactor or add anything to `api.js` but I now use the static functions you mentioned. I considered moving some of the `url-utils.js` functions to `api.js`, but I thought it didn't fit their responsibility of translating between fragments and resources. I'm not 100% certain what all is in the purview of `api.js` but I didn't want to pollute it.

I added `url-utils.test.js` to assist with the refactor and I also went ahead and added a test indicated as missing by a `TODO` in `api.test.js`. I also took the liberty of wrapping the existing tests in a `describe`. I hope I'm not overstepping any boundaries here.